src/app/services/error-handler.service.ts

```diff
@@ -0,0 +1,261 @@
+import { HttpErrorResponse } from '@angular/common/http';
+import { ErrorHandler, Injectable, Injector } from '@angular/core';
+import { AbstractControl, UntypedFormGroup } from '@angular/forms';
+import { TranslateService } from '@ngx-translate/core';
+import * as Sentry from '@sentry/angular';
+import { Observable } from 'rxjs';
+import { ResponseErrorType } from 'app/enums/response-error-type.enum';
+import { sentryCustomExceptionExtraction } from 'app/helpers/error-parser.helper';
+import { ErrorReport } from 'app/interfaces/error-report.interface';
+import { Job } from 'app/interfaces/job.interface';
+import { WebsocketError } from 'app/interfaces/websocket-error.interface';
+import { DialogService } from 'app/services';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ErrorHandlerService implements ErrorHandler {
+  private dialogService: DialogService;
+  private translateService: TranslateService;
+  get translate(): TranslateService {
+    if (!this.translateService) {
+      this.translateService = this.injector.get(TranslateService);
+    }
+    return this.translateService;
+  }
+
+  get dialog(): DialogService {
+    if (!this.dialogService) {
+      this.dialogService = this.injector.get(DialogService);
+    }
+    return this.dialogService;
+  }
+  constructor(private injector: Injector) { }
+
+  handleError(error: unknown): void {
+    console.error(error);
+    let errors: ErrorReport | ErrorReport[];
+    if (this.isTypeOfWebsocketError(error)) {
+      errors = this.parseWsError(error);
+      Sentry.captureException(errors);
+      this.dialog?.error(errors);
+      return;
+    }
+    if (this.isTypeOfJobError(error)) {
+      errors = this.parseJobError(error);
+      Sentry.captureException(errors);
+      this.dialog?.error(errors);
+      return;
+    }
+    sentryCustomExceptionExtraction(error);
+  }
+
+  isTypeOfWebsocketError(error: unknown): error is WebsocketError {
+    return typeof error === 'object'
+      && 'error' in error
+      && 'extra' in error
+      && 'reason' in error
+      && 'trace' in error;
+  }
+
+  isTypeOfJobError(obj: unknown): obj is Job {
+    return typeof obj === 'object'
+    && ('state' in obj
+      && 'error' in obj
+      && 'exception' in obj
+      && 'exc_info' in obj);
+  }
+
+  /**
+   * @param error
+   * @param formGroup
+   * @param fieldsMap Overrides backend field names with frontend field names.
+   * TODO: See if second `string` in fieldsMap can be typed to key of formGroup.
+   */
+  handleWsFormError(
```

Comment:
Will we keep this error handler here or will it belong to a separate FormErrorHandlerService. I feel like latter may be preferrable.